### PR TITLE
test: add EVP_Q_ tests to evp_test

### LINKED ISSUE #15837: Add tests for EVP_Q_* functions
Recently, new public functions have been added to improve usability:

- PR #14664 added `EVP_Q_mac()`
- PR #14741 added `EVP_Q_digest()`

Both were merged, but we missed they did not include tests for the new functions.
This is partially mitigated by the fact that they act as wrappers around the underlying functionality which has tests.

We should add the missing tests for the high level functionality, also as a way to exercise in more combinations the coverage of the underlying functions.

### PR DESCRIPTION
**Pull request by paulidale**
Fixes #15837

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/evp_test.c [hunk 1]
```diff
@@ -476,6 +480,24 @@ static int digest_test_run(EVP_TEST *t)
...
+            && !TEST_ptr(inbuf = sk_EVP_TEST_BUFFER_value(expected->input, 0))
+            && !inbuf->count_set) {
+        OPENSSL_cleanse(got, sizeof(got));
+        if (!TEST_true(EVP_Q_digest(libctx,
+                                    EVP_MD_get0_name(expected->fetched_digest),
+                                    NULL, inbuf->buf, inbuf->buflen,
+                                    got, &size))
+                || !TEST_mem_eq(got, size,
+                                expected->output, expected->output_len)) {
+            t->err = "EVP_Q failed";
```
Comment (levitte) [inline test/evp_test.c:496]: EVP_Q_digest, olease

File: test/evp_test.c [hunk 2]
```diff
@@ -1516,6 +1540,21 @@ static int mac_test_run_mac(EVP_TEST *t)
...
+    if (!xof) {
+        OPENSSL_cleanse(got, sizeof(got));
+        if (!TEST_true(EVP_Q_mac(libctx, expected->mac_name, NULL,
+                                 expected->alg, params,
+                                 expected->key, expected->key_len,
+                                 expected->input, expected->input_len,
+                                 got, got_len, &size))
+                || !TEST_mem_eq(got, size,
+                                expected->output, expected->output_len)) {
+            t->err = "EVP_Q failed";
```
Comment (levitte) [inline test/evp_test.c:1554]: EVP_Q_mac, plesse

File: test/evp_test.c (hunk 1 above)
Comment (paulidale) [reply test/evp_test.c]: I figure reusing the error made checking easier -- this isn't user visible.

Still, no objection to changing it either.

**Review by DDvO — APPROVED** [review]
LGTM

**Review by romen — APPROVED** [review]
LGTM as well!

I do like the idea of testing both this way to bind the 2 interfaces!

**Comment by paulidale** [comment]
Merged, thanks for the reviews.

**Pull request closed without merging**